cli/src/common/infer-protocol.ts

```diff
@@ -0,0 +1,11 @@
+export default function inferProtocol(proxyBaseUrl: string): "http" | "https" {
+  const [protocol] = proxyBaseUrl && proxyBaseUrl.split("://");
+
+  if (protocol !== "http" && protocol !== "https") {
+    throw new Error(
+      'Err - could not infer protocol from proxy base url, should be either "http" or "https".'
```

Comment:
I don't think `Err - ` is necessary since it should be fairly clear in the stack trace that it's an error already?

```suggestion
      'Could not infer protocol from proxy base url, should be either "http" or "https".'
```